src/Microsoft.Data/PrimitiveDataFrameColumn.Sort.cs

```diff
@@ -13,13 +13,13 @@ namespace Microsoft.Data
     public partial class PrimitiveDataFrameColumn<T> : DataFrameColumn
         where T : unmanaged
     {
-        public override DataFrameColumn Sort(bool ascending = true)
+        public new PrimitiveDataFrameColumn<T> Sort(bool ascending = true)
         {
-            PrimitiveDataFrameColumn<long> sortIndices = GetAscendingSortIndices() as PrimitiveDataFrameColumn<long>;
-            return Clone(sortIndices, !ascending, NullCount);
+            PrimitiveDataFrameColumn<long> sortIndices = GetAscendingSortIndices();
+            return Clone(sortIndices, !ascending, NullCount) as PrimitiveDataFrameColumn<T>;
```

Comment:
Do you need the `as` here? Don't we have a Clone method that returns the strong type?